### ISSUE TITLE
Implementation of softmax

### ISSUE THREAD

**Issue by aben20807** [aben20807/blog-post#59]
```c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

double f(const double x) {
    // return x * x;
    return exp(x);
}

void softmax(double* in, double *out, const int len) {
    double f_sum = 0;
    for (int i = 0; i < len; i++) {
        f_sum += f(in[i]);
    }
    for (int i = 0; i < len; i++) {
        out[i] = f(in[i]) / f_sum;
    }
}

int main(int argc, char *argv[]){
    int len = 10;
    double in[] = {
        -34.958106994629, -2.143663406372, -10.308191299438,
         40.310928344727, -20.129846572876, -1.115688323975,
        -30.430717468262, -23.286863327026, -6.127271652222,
        -12.402419090271};
    double *out = calloc(len, sizeof(double));
    softmax(in, out, len);
    for (int i = 0; i < len; i++) {
        printf("%lf\n", out[i]);
    }
    printf("\n");
    return 0;
}
```

**Comment by aben20807** [comment]
[Efficient softmax approximation for GPUs](https://arxiv.org/pdf/1609.04309.pdf)
[On the Computational Efficiency of Training Neural Networks](https://papers.nips.cc/paper/5267-on-the-computational-efficiency-of-training-neural-networks.pdf)
[Alternatives to Softmax for prediction : tensorflow - Reddit](https://www.reddit.com/r/tensorflow/comments/8j1ckk/alternatives_to_softmax_for_prediction/)
